src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializer.java

```diff
@@ -367,6 +367,15 @@ public Object deserializeFromObject(JsonParser p, DeserializationContext ctxt) t
         final Object bean = _valueInstantiator.createUsingDefault(ctxt);
         // [databind#631]: Assign current value, to be accessible by custom deserializers
         p.setCurrentValue(bean);
+
+        // [databind#3838]: Uniform handling of missing objectId
+        if (_objectIdReader != null) {
+            // check if there are any properties present in the JSON object
+            if (!p.hasCurrentToken() || p.getCurrentToken() == JsonToken.END_OBJECT) {
```

Comment:
Actually... no, I don't think this is right. Where does this logic come from? Is it based on similar pattern somewhere, or just trying to see if it resolves specific failing test case?

Reply:
@JooHyukKim Let me try to go back to code -- maybe I am confused.
There are 2 parts to Object Id handling basically -- inclusion of Object Id as part of JSON Object (additional property); and then Object Id Reference (either scalar id or, in very limited cases, single-propert Object). So I want to make sure I understand which parts are being processed.

I'll add another note wrt timing/